k9mail/src/main/java/com/fsck/k9/activity/FolderList.java

```diff
@@ -615,17 +615,18 @@ private void onOpenFolder(String folder) {
         if (folder.equals(allMessagesFolderName)) {
             SearchAccount allMessagesAccount = SearchAccount.createAllMessagesAccount(this);
             MessageList.actionDisplaySearch(this, allMessagesAccount.getRelatedSearch(), false, false, true);
-            return;
-        }
-        if (folder.equals(unifiedFolderName)) {
+        } else if (folder.equals(unifiedFolderName)) {
             SearchAccount unifiedInboxAccount = SearchAccount.createUnifiedInboxAccount(this);
             MessageList.actionDisplaySearch(this, unifiedInboxAccount.getRelatedSearch(), false, false, true);
-            return;
+        } else {
+            LocalSearch search = new LocalSearch(folder);
+            search.addAccountUuid(mAccount.getUuid());
+            search.addAllowedFolder(folder);
+            MessageList.actionDisplaySearch(this, search, false, false, true);
+        }
+        if (BuildConfig.IS_ENTERPRISE) {
+            finish();
```

Comment:
it seems like a feature for non-enterprise versions only, isn't it?

Reply:
Yes, the Folder List, which we don't want to be accessible in enterprise variant (but it can still be accessed in some corner cases and that should be addressed at some point).